stdlib/result.gr

```diff
@@ -1,89 +1,203 @@
+    Err(e) => Err(e),
+  }
 }
 
-export let flatMapErr = (fn, v) => {
-    match (v) {
-        Err(e) => fn(e),
-        Ok(x)  => Ok(x)
-    }
+/**
+ * If the Result is an `Err(value)`, applies the given function to the `value` to produce a new Result.
+ *
+ * @param fn: The function to call on the value of an `Err` variant
+ * @param result: The result to map
+ * @returns A new Result produced by the mapping function if the variant was `Err` or the unmodified `Ok` otherwise
+ */
+export let flatMapErr = (fn, result) => {
+  match (result) {
+    Err(e) => fn(e),
+    Ok(x) => Ok(x),
+  }
 }
 
-export let map = (fn, v) => {
-    match (v) {
-        Ok(x)  => Ok(fn(x)),
-        Err(e) => Err(e)
-    }
+/**
+ * If the Result is `Ok(value)`, applies the given function to the `value` and wraps the new value in an `Ok` variant.
+ *
+ * @param fn: The function to call on the value of an `Ok` variant
+ * @param result: The result to map
+ * @returns A new `Ok` variant produced by the mapping function if the variant was `Ok` or the unmodified `Err` otherwise
+ */
+export let map = (fn, result) => {
+  match (result) {
+    Ok(x) => Ok(fn(x)),
+    Err(e) => Err(e),
+  }
 }
 
-export let mapErr = (fn, v) => {
-    match (v) {
-        Err(e) => Err(fn(e)),
-        Ok(x)  => Ok(x)
-    }
+/**
+ * If the Result is `Err(value)`, applies the given function to the `value` and wraps the new value in an `Err` variant.
+ *
+ * @param fn: The function to call on the value of an `Err` variant
+ * @param result: The result to map
+ * @returns A new `Err` variant produced by the mapping function if the variant was `Err` or the unmodified `Ok` otherwise
+ */
+export let mapErr = (fn, result) => {
+  match (result) {
+    Err(e) => Err(fn(e)),
+    Ok(x) => Ok(x),
+  }
 }
 
-export let mapWithDefault = (fn, def, v) => {
-    match (v) {
-        Ok(a) => fn(a),
-        _     => def
-    }
+/**
+ * If the Result is `Ok(value)`, applies the given function to the `value` to produce a new value, otherwise uses the default value.
+ * Useful for unwrapping a successful Result while providing a fallback for any errors that can occur.
+ *
+ * @param fn: The function to call on the value of an `Ok` variant
+ * @param def: A fallback value for an `Err` variant
+ * @param result: The result to map
+ * @returns The value produced by the mapping function if the result is of the `Ok` variant or the default value otherwise 
+ */
+export let mapWithDefault = (fn, def, result) => {
+  match (result) {
+    Ok(a) => fn(a),
+    _ => def,
+  }
 }
 
-export let mapWithDefaultFn = (fnOk, fnErr, v) => {
-    match (v) {
-        Ok(a)  => fnOk(a),
-        Err(a) => fnErr(a)
-    }
+/**
+ * If the Result is `Ok(value)`, applies the `fnOk` function to the `value` to produce a new value. If the Result is `Err(value)`, applies the `fnErr` function to the `value` to produce a new value.
+ * Useful for unwrapping a Result into a value, whether it is successful or unsuccessful.
```

Comment:
```suggestion
 * If the Result is `Ok(value)`, applies the `fnOk` function to the `value` to produce a new value.
 * If the Result is `Err(value)`, applies the `fnErr` function to the `value` to produce a new value.
 * Useful for unwrapping a Result into a value, whether it is successful or unsuccessful.
```
Sorry, another really long line

Reply:
Quick question on this wont the website and vs code and other editors wrap this naturally if not is there a max length that you think i should go to or just break it up into paragraphs.

Reply:
Yeah, they will wrap. I just feel like a natural place to break is on period.

This only applies to descriptions, as all other docs must go on a single line due to my very lazy regexp parser.